diff --git a/cctools/ld64/src/ld/OutputFile.cpp b/cctools/ld64/src/ld/OutputFile.cpp
index cc12ba1..c03c605 100644
--- a/cctools/ld64/src/ld/OutputFile.cpp
+++ b/cctools/ld64/src/ld/OutputFile.cpp
@@ -3998,7 +3998,7 @@ void OutputFile::writeOutputFile(ld::Internal& state)
 	}
 
 	// assume mappable by default
-	bool outputIsMappableFile = true;
+	bool outputIsMappableFile = _options.architecture() != CPU_TYPE_ARM64;
 
 #if __arm64__
 	// <rdar://problem/66598213> work around VM limitation on Apple Silicon and use write() instead of mmap() to produce output file
