From 875ab8090b8d645b4a9ca7e60753b4726959156a Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Tue, 14 Mar 2023 17:36:17 +1100
Subject: [PATCH 5/5] don't try to include __assert for NDEBUG builds

---
 cctools/ld64/src/3rd/helper.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cctools/ld64/src/3rd/helper.c b/cctools/ld64/src/3rd/helper.c
index 179a233..37b0e2b 100644
--- a/cctools/ld64/src/3rd/helper.c
+++ b/cctools/ld64/src/3rd/helper.c
@@ -37,6 +37,7 @@ const char ldVersionString[] = "@(#)PROGRAM:ld  PROJECT:ld64-" STRINGIFY(LD64_VE
 
 void __assert_rtn(const char *func, const char *file, int line, const char *msg)
 {
+#ifndef NDEBUG
 #if defined(__FreeBSD__) || defined(__DragonFly__)
     __assert(msg, file, line, func);
 #elif defined(__NetBSD__) || defined(__OpenBSD__) || defined(__CYGWIN__)
@@ -48,6 +49,7 @@ void __assert_rtn(const char *func, const char *file, int line, const char *msg)
     fflush(NULL);
     abort();
 #endif /* __FreeBSD__ */
+#endif /* NDEBUG */
 }
 
 int _NSGetExecutablePath(char *epath, unsigned int *size)
-- 
2.38.1.windows.1

